File: src/Card.java
public class Card {

    private int number;
    private Suit suit;

    public Card(int number, Suit suit) {
        this.number = number;
        this.suit = suit;
    }

    public int getNumber() {
        return number;
    }

    public Suit getSuit() {
        return suit;
    }

    public Color getColor() {
        switch (this.getSuit()) {
            case HEARTS:
            case DIAMONDS:
                return Color.RED;
            case SPADES:
            case CLUBS:
                return Color.BLACK;
            default:
                throw new IllegalStateException("Unexpected value: " + this.getSuit());
        }
    }

    public String getCardName(){
        String prefix;
        switch (this.getSuit()){
            case SPADES:
                prefix = "S";
                break;
            case HEARTS:
                prefix = "H";
                break;
            case DIAMONDS:
                prefix = "D";
                break;
            case CLUBS:
                prefix = "C";
                break;
            default:
                prefix = "";
        }

        return prefix + this.getNumber();
    }

}

File: card.py
from enum import Enum


class Suit(Enum):
    DIAMONDS = 1
    CLUBS = 2
    SPADES = 3
    HEARTS = 4


class Color(Enum):
    RED = 1
    BLACK = 2


class Card:
    ACE = 1

    def __init__(self, number, suit):
        self.suit = suit
        self.number = number

    def name(self):
        switch = {
            Suit.HEARTS: 'H',
            Suit.DIAMONDS: 'D',
            Suit.CLUBS: 'C',
            Suit.SPADES: 'S'
        }
        return switch.get(self.suit, None) + str(self.number)

    def color(self):
        if self.suit in (Suit.HEARTS, Suit.DIAMONDS):
            return Color.RED
        elif self.suit in (Suit.CLUBS, Suit.SPADES):
            return Color.BLACK
        else:
            return None
